### ISSUE TITLE
Why does Phoenix.Component not support :global attr  and :default option within a :slot?

### ISSUE THREAD

**Issue by c4710n** [phoenixframework/phoenix_live_view#3218]
> First of all, I apologize for posting my question here; I know I should post this kind of content on ElixirForum.
>
> I posted [a question on ElixirForum](https://elixirforum.com/t/why-does-phoenix-component-not-support-global-attr-within-a-slot/63026) but did not receive any answer. So, I'm posting my question here in hopes of getting some responses from core developers.

I've always wanted **top-level attrs** and **attrs in slots** to have the same interface:
+ can use `:global` type.
+ use same available options.

But, currently:
+ attrs in slots can't use `:global` type
+ attrs in slots can't use `:default` option

---

I was unclear about the reasons, so I tried to do some research. What I can find are:
+ [the issue that we shoudn't support :global attr within a :slot](https://github.com/phoenixframework/phoenix_live_view/issues/2164)
+ [a PR related to above issue](https://github.com/phoenixframework/phoenix_live_view/pull/2294)
+ [and, some discussions about this issue](https://github.com/phoenixframework/phoenix_live_view/issues/2265)

> And, I also tried to reading the code, and I might have found some clues, but I'm not quite sure. Such as:
> + https://github.com/plastic-forks/phoenix_live_view/blob/7e885f34e8803a719f54c8d1acb2ede10ba5fdac/lib/phoenix_component/declarative.ex#L642

Beyond above links, I can't find other useful information.

I'm looking forward to some explanations from core developers, be it in-depth discussions or just a word or two, anything would be great.

**Comment by chrismccord** [comment]
> I'm looking forward to some explanations from core developers, be it in-depth discussions or just a word or two, anything would be great.

tldr; It would introduce a lot of complexity and performance penalty at compile time
